osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -190,7 +190,9 @@ protected override void LoadComplete()
             comboIndexBindable.BindValueChanged(_ => UpdateComboColour());
             comboIndexWithOffsetsBindable.BindValueChanged(_ => UpdateComboColour(), true);
 
-            updateState(ArmedState.Idle, true);
+            // If the state is changed, transforms are already initialized.
+            if (state.Value == ArmedState.Idle)
```

Comment:
Adding this guard is the same as specifying `force: false` in `updateState()`, is it not? Which makes me wonder why the true was there in the first place and whether doing this won't break something else.

Reply:
It is the opposite. `force: false` will apply transforms when the current state is *not* `Idle`.

Reply:
...which is different from the guard how? I'm confused. The guard is checking for idle, and not applying transforms forcefully if idle...?

Reply:
Never mind, you're correct.

Reply:
It reads backwards though, the rare case is when state changes before load, but most of the time it wouldn't. An early-return would make it read better:
```csharp
// The state has already been changed, no need to initialize transforms.
if (state.Value != ArmedState.Idle)
    return;

updateState(ArmedState.Idle, true);
```

Reply:
This change is obsolete.